packages/observable-client/src/getObservableClient.ts

```diff
@@ -19,6 +19,9 @@ const clientCache = new Map<
 
 export const getObservableClient = (
   substrateClient: SubstrateClient,
+  getCachedMetadata: (codeHash: string) => Observable<Uint8Array | null> = () =>
+    of(null),
+  setCachedMetadata: (codeHash: string, rawMetadata: Uint8Array) => void = noop,
```

Comment:
I'd prefer if this was also part of an "options" parameter.... I know the observable client API is unstable, but I'm never a fan of things that are options as part of the ordered arguments.